eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java

```diff
@@ -138,18 +142,150 @@ private int optimized(final ForeignTojo tojo, final Optimization optimization)
+     */
+    public static final class OptTrain implements Function<XML, XML> {
+
+        /**
+         * Parsing train with XSLs.
+         * @implNote The list of applied XSLs is adjusted during execution.
+         * <br>Separate instance of the train is used of each optimization
+         * thread since {@link com.jcabi.xml.XSLDocument}, which is used under
+         * the hood in {@link com.yegor256.xsline.TrClasspath}, is not thread-safe.
+         */
+        static final Train<Shift> DEFAULT_TRAIN = new TrFast(
+            new TrLambda(
+                new TrClasspath<>(
+                    "/org/eolang/parser/optimize/globals-to-abstracts.xsl",
+                    "/org/eolang/parser/optimize/remove-refs.xsl",
+                    "/org/eolang/parser/optimize/abstracts-float-up.xsl",
+                    "/org/eolang/parser/optimize/remove-levels.xsl",
+                    "/org/eolang/parser/add-refs.xsl",
+                    "/org/eolang/parser/optimize/fix-missed-names.xsl",
+                    "/org/eolang/parser/add-refs.xsl",
+                    "/org/eolang/parser/set-locators.xsl",
+                    "/org/eolang/parser/blank-xsd-schema.xsl"
+                ).back(),
+                StEoLogged::new
+            ),
+            TrFast.class,
+            500L
+        );
+
+        /**
+         * Delegate.
+         */
+        private final Function<XML, XML> delegate;
+
+        /**
+         * Shifts that we are going to apply.
+         */
+        private final Train<Shift> shifts;
+
+        /**
+         * The default constructor with the default preset of xsl optimizations.
+         */
+        public OptTrain() {
+            this(OptTrain.DEFAULT_TRAIN);
+        }
+
+        /**
+         * Constructor that accepts train of shifts.
+         * @param shifts XLS shifts.
+         */
+        public OptTrain(final Train<Shift> shifts) {
+            this(xml -> xml, shifts);
+        }
+
+        /**
+         * Constructor for single {@link com.yegor256.xsline.StClasspath} optimization.
+         * @param delegate Optimizations that have to be done before.
+         * @param xls File from classpath.
+         */
+        public OptTrain(final Function<XML, XML> delegate, final String xls) {
+            this(
+                delegate,
+                new TrDefault<Shift>().with(new StClasspath(xls))
+            );
+        }
+
+        /**
+         * The default constructor.
+         * @param delegate Optimizations that have to be done before.
+         * @param shifts To apply
+         */
+        public OptTrain(
+            final Function<XML, XML> delegate,
+            final Train<Shift> shifts
+        ) {
+            this.delegate = delegate;
+            this.shifts = shifts;
+        }
+
+        @Override
+        public XML apply(final XML xml) {
+            return new Xsline(this.shifts).pass(this.delegate.apply(xml));
```

Comment:
@ArtemGet I think instead on creating `new Xsline` on every `apply` call, we can create it only once in constructor and encapsulate it

Reply:
It's a good point, done.